crates/firewheel-core/src/node.rs

```diff
@@ -28,12 +28,13 @@ impl Default for NodeID {
 ///
 /// This struct enforces the use of the builder pattern for future-proofness, as
 /// it is likely that more fields will be added in the future.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug)]
 pub struct AudioNodeInfo {
     debug_name: &'static str,
     channel_config: ChannelConfig,
     uses_events: bool,
     call_update_method: bool,
+    custom_state: Option<Box<dyn Any>>,
```

Comment:
This is super nitpicky, so take it for what it's worth, but could we possibly separate out the state from the info struct?

The reason I ask is that _technically_, a user could provide the state in both `AudioNode::info` and `AudioNode::processor`, although it seems like it's supposed to be constructed in the latter (correct me if I'm wrong).

Reply:
The same `Option<Box<dyn Any>>` object that is created in the `AudioNodeInfo` struct is what is passed to `AudioNode::processor` and `AudioNode::update`. The node is free to mutate these at any point, but generally nodes should construct it in `Info` so that the state is immediately available to the user in `